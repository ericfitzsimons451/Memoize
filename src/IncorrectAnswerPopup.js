import React from 'react';
import './IncorrectAnswerPopup.scss';

function IncorrectAnswerPopup(props) {
  return(
    <div className="popup-container">
      <h2 className="popup-title"></h2>
      <h3>Sorry!  That answer is not correct.  Please try again.</h3>
      <button className="popup-button" type="button" onClick={props.returnToDisplayAll}>Return to Question List</button>
    </div>
    )
}

export default IncorrectAnswerPopup;