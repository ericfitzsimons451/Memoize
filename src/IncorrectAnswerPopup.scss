.popup--container {
  width: 600px;
  height: 450px;
  border: 1px solid black;
  background-color: #778078;
  background-image: url("https://www.transparenttextures.com/patterns/asfalt-dark.png");
  display: flex;
  flex-direction: column;
  margin: 5px;
  border-radius: 15px;
  justify-content: space-around;
}

.popup--title {
  font-size: 2.3rem;
}

.popup-text {
  font-size: 2.3rem;
}

.popup--button {
  height: 80px;
  font-size: 1.7rem;
  margin: 0px 20px 0px 20px;
  padding: 3px;
  background-color: white;
  border-radius: 15px;
  &:hover {
    cursor: pointer;
  }
}