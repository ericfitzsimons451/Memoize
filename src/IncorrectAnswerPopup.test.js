import React from 'react';
import IncorrectAnswerPopup from './IncorrectAnswerPopup.js';
import { shallow } from 'enzyme';

const mockReturnToDisplayAll = jest.fn();

describe('IncorrectAnswerPopup', () =>{
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <IncorrectAnswerPopup returnToDisplayAll={mockReturnToDisplayAll} />
      )
  })

  it('should match the snapshot with all correct data passed down', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should register a click when RETURN TO QUESTION LIST has been clicked', () => {
    wrapper.find('.popup--button').simulate('click')
    expect(mockReturnToDisplayAll).toBeCalled()
  })
})