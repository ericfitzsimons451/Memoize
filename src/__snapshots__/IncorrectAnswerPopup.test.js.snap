// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncorrectAnswerPopup should match the snapshot with all correct data passed down 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IncorrectAnswerPopup
    returnToDisplayAll={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="popup-title"
        />,
        <h3>
          Sorry!  That answer is not correct.  Please try again.
        </h3>,
        <button
          className="popup-button"
          onClick={[MockFunction]}
          type="button"
        >
          Return to Question List
        </button>,
      ],
      "className": "popup-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "popup-title",
        },
        "ref": null,
        "rendered": null,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sorry!  That answer is not correct.  Please try again.",
        },
        "ref": null,
        "rendered": "Sorry!  That answer is not correct.  Please try again.",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Return to Question List",
          "className": "popup-button",
          "onClick": [MockFunction],
          "type": "button",
        },
        "ref": null,
        "rendered": "Return to Question List",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="popup-title"
          />,
          <h3>
            Sorry!  That answer is not correct.  Please try again.
          </h3>,
          <button
            className="popup-button"
            onClick={[MockFunction]}
            type="button"
          >
            Return to Question List
          </button>,
        ],
        "className": "popup-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "popup-title",
          },
          "ref": null,
          "rendered": null,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sorry!  That answer is not correct.  Please try again.",
          },
          "ref": null,
          "rendered": "Sorry!  That answer is not correct.  Please try again.",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Return to Question List",
            "className": "popup-button",
            "onClick": [MockFunction],
            "type": "button",
          },
          "ref": null,
          "rendered": "Return to Question List",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
